source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -272,6 +283,7 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(
 
   for (auto upgrade_config : config.upgrade_configs()) {
     const std::string& name = upgrade_config.upgrade_type();
+    bool enabled = upgrade_config.has_enabled() ? upgrade_config.enabled().value() : true;
     if (findUpgradeCaseInsensitive(upgrade_filter_factories_, name) !=
```

Comment:
a thought: it would be really cool if we had a pgv constraint for repeated fields for unique key in the repeated message. That way we wouldn't have to implement checks like this here.

cc @htuch

Reply:
@junr03 you mean like this one https://github.com/lyft/protoc-gen-validate/blob/1ffd5e8db575ed8a61c7e0d415af263c39cef1ab/validate/validate.proto#L642? :D

Reply:
I should have known that was already in place. @rodaine has a tendency for building what is needed :)

Reply:
If a constraint exists would this be a good opportunity to make the change to use it?

Reply:
Unfortunately that's not the constraint we're looking for, given the case sensitivity issue
If we want to half-enforce in constraints and half in code we can do that but seems odd to me.

Reply:
OK no problem

Reply:
I see, right, the available constraint is more restrictive than what we want here. I'll follow up with @rodaine. Maybe we can make this constraint more expressive for strings to allow for case-sensitivity.

Reply:
That'd be nifty.  If it lands feel free to tag me for a clean up PR!